### ISSUE TITLE
Enable using Hyper-V Default Switch for Fall Creator's update (Windows 10)

### ISSUE THREAD

**Issue by gbraad** [minishift/minishift#1792]
Since the latest update of Windows 10, the Fall Creator's update, Hyper-V provides an easy networking option by offering a "Default Switch" which does DNS, DHCP (, and NAT).

---

The switch named “Default Switch” allows virtual machines to share the host’s network connection using NAT (Network Address Translation).  This switch has a few unique attributes:

 1. Virtual machines connected to it will have access to the host’s network whether you’re connected to WIFI, a dock, or Ethernet. It will also work when the host is using VPN
    or proxy.
 2. It’s available as soon as you enable Hyper-V – you won’t lose internet setting it up.
 3. You can’t delete or rename it.
 4. It has the same name and device ID on all Windows 10 Fall Creator’s Update Hyper-V hosts.
     * Name: `Default Switch`
     * ID: `c08cb7b8-9b3c-408e-8e30-5e16a3aeb444`

---

Ref: https://blogs.technet.microsoft.com/virtualization/2017/11/13/whats-new-in-hyper-v-for-windows-10-fall-creators-update/, https://github.com/kubernetes/minikube/issues/1627#issuecomment-348934700

**Comment by gbraad** [comment]
This nullifies the effort to create a switch for #1676, and changes the default name. (Note: the limitation of one NAT network remains). So, if you have created a NAT network, the default network will not be created.

**Comment by gbraad** [comment]
@robin-owen I think we need to change some of the current documentation to reflect the chanegs between the Windows 10 Creator's Update (1703) and Windows 10 Fall Creator's update (1709). Any suggestions?

**Comment by robin-owen** [comment]
@gbraad I'm not familiar with the way that Windows handles these updates. I'm assuming that not everyone will have the Fall Creator's Update, so having documentation for both makes sense to me (at least until we can reasonably assume that everyone has the latest update).

**Comment by gbraad** [comment]
So far, the updates get forced and after a while they should be generally assumed. However, I have one situation where the updates fails to apply (and rolls back). So we have to describe both cases. However, I am not sure about the request we original saw in issue #1676 where a default name of "minishift" was suggested. I'd rather not have in code to accomodate for `minishift` as this is not the default going forward, but in the documentation? /cc: @minishift/minishift-dev WDYT?

**Comment by gbraad** [comment]
To use:

```
PS> $env:HYPERV_VIRTUAL_SWITCH="Default Switch"
PS> minishift start
```

This will assign an address like `172.26.219.100/20`, and sets `resolv.conf` with 
```
search mshome.net
```

However, this does not seem to work for me. No address are being resolved. Maybe because I already have a NAT network assigned to `DockerNAT` ?

**Comment by gbraad** [comment]
Startup works, until doing anything related to DNS. This fails due to:

```
search mshome.net
```

**Comment by gbraad** [comment]
Related issue: https://github.com/docker/for-win/issues/1166#issuecomment-340062138

**Comment by anjannath** [comment]
minishift works properly following the instructions on this [comment](https://github.com/minishift/minishift/issues/1792#issuecomment-349704223), when _docker for windows (Stable)_ is installed.

The issue appears when _docker for windows (Edge)_ is installed and LCOW is enabled.

**Comment by gbraad** [comment]
```
func AddNameserverToInstance(driver drivers.Driver, nameserver string) {
	executeCommandOrExit(driver,
		fmt.Sprintf("NS=%s; cat /etc/resolv.conf |grep -i \"^nameserver $NS\" || echo \"nameserver $NS\" | sudo tee -a /etc/resolv.conf", nameserver),
		"Error adding nameserver")
}
```

Also going to add a preflight check in post to check if a nameserver is configured.

**Comment by gbraad** [comment]
```
PS> $env:HYPERV_VIRTUAL_SWITCH="Default Switch"
PS> .\minishift.exe start --network-nameserver 8.8.8.8 --network-nameserver 8.8.4.4 
```

**Comment by gbraad** [comment]
Using `Get-NetIPInterface -InterfaceAlias "vEthernet (Default Switch)" -AddressFamily IPv4 | Get-NetIPAddress | ForEach-Object { $_.IPAddress }` seems to provide me with the correct address for use as DNS and Gateway for the VMs on the "Default Switch"

**Comment by robin-owen** [comment]
@gbraad Considering I'm working on the installation instructions at the moment, I can add this information to the documentation today or tomorrow if the instructions given here reliably work. Is that the case?

**Comment by LalatenduMohanty** [comment]
Resolved via #1849
